catinabox/catgenerator.py

```diff
@@ -0,0 +1,36 @@
+import random
+import requests
+import time
+
+
+NAME_GENERATOR_API_ENDPOINT = "http://namey.muffinlabs.com/name.json"
+# ignoring leap years for now
+SECONDS_IN_YEAR = 365 * 24 * 60 * 60
+MAX_YEARS_OLD = 30
+
+
+class CouldNotGetNameError(Exception):
+    def __init__(self, e):
+        super(CouldNotGetNameError, self).__init__(
+            "Unable to connect to {}: {}".format(
+                NAME_GENERATOR_API_ENDPOINT, e
+            )
+        )
+
+
+def cat_generator():
+    while True:
+        try:
+            result = requests.get(NAME_GENERATOR_API_ENDPOINT)
+            name = result.json()[0]
+        except requests.exceptions.RequestException as e:
+            raise CouldNotGetNameError(e)
```

Comment:
What do you think about passing in `str(e)` instead? It doesn't look like `CouldNotGetNameError` needs anything in particular from the exception object except for that.